src/apps/statements/repo/modelsRepo/updateFullActivities/mongo.ts

```diff
@@ -1,47 +1,107 @@
 import matchesFullActivity from '../utils/mongoModels/matchesFullActivity';
 import { replaceDotsInExtensions } from '../utils/mongoModels/replaceDotsInStatement';
 import Signature from './Signature';
 
-const getPatchUpdate = <T>(patch: Dictionary<T>, parentKeys: string[]) => {
-  return mapKeys<T>(patch as any, (_value: T, key: number) => {
-    const parentPath = parentKeys.join('.');
-    return `${parentPath}.${key}`;
+const getPatchUpdate = <T>(patch: Dictionary<T>, parentKeys: string[]) =>
+  mapKeys<T>(
+    patch as any,
+    (_value: T, key: number) => `${parentKeys.join('.')}.${key}`,
+  );
+
+const createMongoAddToSet = (fullActivity: FullActivityDatabase) => (
+  {
+    ...(fullActivity.context?.contextActivities?.parent !== undefined
+        ? {
+          'context.contextActivities.parent': {
+            $each: fullActivity.context.contextActivities.parent,
+          },
+        }
+        : {}
+    ),
+    ...(fullActivity.context?.contextActivities?.category !== undefined
+        ? {
+          'context.contextActivities.category': {
+            $each: fullActivity.context.contextActivities.category,
+          },
+        }
+        : {}
+    ),
+    ...(fullActivity.context?.contextActivities?.other !== undefined
+        ? {
+          'context.contextActivities.other': {
+            $each: fullActivity.context.contextActivities.other,
+          },
+        }
+        : {}
+    ),
+    ...(fullActivity.context?.contextActivities?.grouping !== undefined
+        ? {
+          'context.contextActivities.grouping': {
+            $each: fullActivity.context.contextActivities.grouping,
+          },
+        }
+        : {}
+    ),
+  }
+);
+
+const createMongoSet = (fullActivity: FullActivityDatabase) => {
+  const extensions = replaceDotsInExtensions(/\./g, '&46;')(fullActivity.extensions);
+
+  return {
+    ...getPatchUpdate(fullActivity.name, ['name']),
+    ...getPatchUpdate(fullActivity.description, ['description']),
+    ...getPatchUpdate(extensions, ['extensions']),
+    ...(
+      fullActivity.moreInfo !== undefined
+        ? { moreInfo: fullActivity.moreInfo }
+        : {}
+    ),
+    ...(
+      fullActivity.type !== undefined
+        ? { type: fullActivity.type }
+        : {}
+    ),
+  };
+};
+
+const createMongoQuery = (fullActivity: FullActivityDatabase) => {
+  const activityId = fullActivity.activityId;
+  const lrsId = new ObjectID(fullActivity.lrsId);
+  const organisationId = new ObjectID(fullActivity.organisationId);
+
+  return matchesFullActivity({ activityId, lrsId, organisationId });
+};
+
+const creatBatchQuery = (batch: UnorderedBulkOperation) => (fullActivity: FullActivityDatabase) => {
+  const mongoQuery = createMongoQuery(fullActivity);
+  // tslint:disable:no-inferred-empty-object-type
+  const mongoSet = createMongoSet(fullActivity);
+  const mongoAddToSet = createMongoAddToSet(fullActivity);
+  // tslint:enable:no-inferred-empty-object-type
```

Comment:
@PrinceWaune we don't use such ignore lines in other modules within the xapi-service - is this required?